DMTN-148.tex

```diff
@@ -50,9 +50,276 @@
+product system.  For FITS image products, this requires future
+calibrations to be constructed with all of the header keyword values
+listed above.
+
+The remaining unconverted calibration types (\verb|CROSSTALK|, \verb|BFKERNEL|)
+should be converted to use \verb|DEFECTS|-like class, most likely by defining
+a CalibrationBaseClass, and making all of the calibration products
+subclasses of that.  This would enforce that all calibrations can be
+relied upon to provide getMetadata, setMetadata, readFits, writeFits,
+readText, and writeText methods.  Gen2 Butler instances would then be
+able to access all calibration products in a single consistent manner.
+This also ensures that Gen2 and Gen3 can use the same calibration
+products.
+
+\subsection{Metadata and Header Keywords}
+
+The required header keywords are currently poorly standarized.  The
+\verb|OBSTYPE| keyword is universally populated in all calibrations,
+regardless of if they are generated by constructCalibs.py or are
+curated by the \verb|obs_CAMERA| packages.  Curated calibrations also
+require the \verb|DETECTOR| and \verb|CALIBDATE|, the last of which is
+also populated by constructCalibs.py.  The \verb|CALIB_ID| is set by
+constructCalibs.py, and exists for the majority of curated
+calibrations, but is not consistently used.
+
+The keywords listed in table \ref{tab:metadata} are the proposed list
+of header keywords that will be required in the new operational
+calibrations in Gen3.  The majority of these are inserted at creation,
+identifying the content and applicable data range for the calibration.
+Only upon certification (the replacement for the Gen2 ingest) are the
+valid date ranges set.  To conform with the expected changes to FITS
+persistence in RFC-686, these and all other calibration-specific
+metadata fields will be upper-case.
+
+\begin{tabular}{l l l}
+  \label{tab:metadata}
+  Keyword & Inserted & Description \\
+  \hline
+  \verb|OBSTYPE| & Creation & Calibration dataset type contained in this file. \\
+  \verb|{OBSTYPE}_SCHEMA| & Creation & Storage schema for calibration.  \\
+  \verb|{OBSTYPE}_VERSION| & Creation & Storage schema version. \\
+  \verb|INSTRUME| & Creation & Instrument/camera this calibration belongs to. \\
+  \verb|DETECTOR| & Creation & Integer identifier for the ccd/detector. \\
+  \verb|FILTER| & Creation & Name of the physical filter or ``NONE''. \\
+  \verb|CALIB_ID| & Creation & DataId that can be used to request this calibration after ingest. \\
+  \verb|CALIB_CREATION_DATE| & Creation & Creation date of the calibration. \\
+  \verb|CALIB_CREATION_TIME| & Creation & Creation time of the calibration. \\
+  \verb|CALIBDATE| & Creation & Concatenation of \verb|CALIB_CREATION_DATE| and \verb|CALIB_CREATION_TIME|. \\
+  \verb|VALIDSTART| & Certification & Start date the calibration is valid from.  \\
+  \verb|VALIDEND| & Certification & End date the calibration is valid from.  \\
```

Comment:
Is there is in-place modification of on-disk datasets upon certification?  Or does this mean that on export some headers come from certain inputs and some from others?

Reply:
The data are never touched during certification. The second column states that in this table. In the _data packages we use the file name to show validity range.

Reply:
This table says "Inserted", which suggests modification to me.  Perhaps a different verb should be used, or it should be made clear where these are inserted if not in the dataset.

Reply:
I'll make sure this is clear in the text to represent my claim above that until the file is exported, the validity is handled solely by the butler.